pkg/controller/controller_test.go

```diff
@@ -3114,6 +3100,35 @@ func TestProvisionFromSnapshot(t *testing.T) {
 	}
 }
 
+func (s *testcase) notNilSnapshotStatusTest(notNilSnapshotStatusTest bool) {
```

Comment:
Hi @msau42 , i think added a constructor could be useful while i changing removing if-else statement and using a method instead. i am facing error when fetching `expectedpvSpec` property from  testcase struct.